var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -585,22 +624,43 @@ def install(self, spec, prefix):
         ]
         components["flang-runtime"] += flang_common_args
 
-        build_order = ["aomp-extras", "openmp", "openmp-debug", "pgmath", "flang", "flang-runtime"]
```

Comment:
i think for older releases like 5.7.1 etc the new build order will break. when building for 5.71 i saw below error 

  >> 14936    CMake Error: The source directory "/tmp/root/spack-stage/spack-stage-rocm-openmp-extras-5.7.1-nvfbpsjthzkzntke5ybrmbzbxfozbgzu/spack-src/rocm-openmp-extras/flang/flang-legacy/17.0-4/llvm-legacy/llvm" does not exist.
Can we add if self.spec.satisfies("@:6.0"): and include the older build_order . 
     14937    Specify --help for usage, or press the help button on the CMake GUI.